modules/backend/formwidgets/repeater/assets/js/repeater.js

```diff
@@ -259,7 +259,7 @@
                     return $textInput.val()
             }
         } else {
-            var $disabledTextInput = $('.text-field:first > .form-control', $target)
+            var $disabledTextInput = $('.form-control:first', $target)
```

Comment:
if the `text-field` class is on `$target`, could we do `&.text-field` instead? Also is this currently broken in both update and preview contexts? If so, what is the difference between them that is causing the structure of the fields to be different?

Reply:
In update context the HTML structure is different.
It has an [input field](https://github.com/wintercms/winter/blob/732ca4190efc2269b64fc8f119518e5b70e3ff73/modules/backend/formwidgets/repeater/partials/_repeater.php#L29), which is [selected in JavaScript](https://github.com/wintercms/winter/blob/732ca4190efc2269b64fc8f119518e5b70e3ff73/modules/backend/formwidgets/repeater/assets/js/repeater.js#L251).
titleFrom in this context is already working as described in the docs.

In preview mode however, the HTML structure is different.
It does not have an input field, but a `<span class='form-control'>` instead.
The JavaScript will go in the else part and select the 'disabled text input'.

**Here are actually 2 options:**

1. You defined titleFrom: target is the parent of `<span class='form-control'>`
2. You didn't define titleFrom: target is the row itself

**This is where it goes wrong in preview context.**
In case 1, it will try to select `.text-field:first` but that does not exist because its on the element itself resulting in an empty string.
In case 2, it will find the first `ul` and display that text. In my case this is not my text field, but a balloon selector. Remember, there are no form fields in preview context. input & select are not present.

**So I fixed**
Case 1 by removing the `.text-field:first` because it is an invalid selector. But yes, it could have the &. notation but don't see the advantage of that in this case?
Case 2 will have the same behaviour as before.

Reply:
Do I still need to do something to finish this PR? 🙂

Reply:
@AIC-BV it seems mostly fine, would you be willing to take a stab at writing a test case for this?